sdk/servicebus/Azure.Messaging.ServiceBus/api/Azure.Messaging.ServiceBus.netstandard2.0.cs

```diff
@@ -430,6 +432,22 @@ public partial class ServiceBusSessionReceiverOptions
         [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
         public override string ToString() { throw null; }
     }
+    public sealed partial class ServiceBusSharedAccessKeyCredential
```

Comment:
I think we should split this into two types like we do with other key/signature credentials.

Reply:
Can you provide pointers to the other credentials you are referring to here? Is the reason that one wouldn't think of using this credential if all they had was the signature (because the name says "key")? Would a rename then help? `ServiceBusSharedAccessCredential`? `ServiceBusSharedAccessPolicyCredential`?

Reply:
I'm not sure that I understand what the benefit to callers would be in splitting this.  As these are elements that are often sibling concepts within a connection string, there is precedent for their association and I would prefer to avoid introducing another constructor overload or two additional public types (base class, other credential) into an already crowded space.

Reply:
Storage has a StorageSharedKeyCredential and we're looking at StorageSharedAccessSignatureCredential to match EventGrid's.  And the benefit to splitting it is make auth, a historically difficult topic, as clear as possible.

Reply:
This one I do feel strongly about.   I don't believe that splitting this offers greater usability, and I do believe that the additional types and overloads degrade it.   This, I'd very much like to push back on.